lib/health.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const hasMatchingMethod = (option, reqMethod) => {
+  return option.methods.some(method => {
+    return method.toLowerCase() === reqMethod.toLowerCase();
+  });
+};
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        // Test the method property against the current req method
+        if (option.methods && (hasMatchingMethod(option, req.method) !== true)) {
+          return memo;
+        }
+        // memoize the endpoints
+        memo.push(request(option.url).promise());
+        return memo;
+      }, []);
```

Comment:
I wouldn't add the `steps` parameter. There's already ways of mounting middleware onto particular paths within an app - i.e. `app.use('/path', middleware);`

Alternatively, a user can wrap this in a behaviour and mount on whatever steps they like.

Either way, I'd leave that configuration external to this module, and keep this module small.